pytorch_lightning/trainer/properties.py

```diff
@@ -507,7 +511,13 @@ def min_steps(self) -> Optional[int]:
 
     @property
     def callback_metrics(self) -> dict:
-        return self.logger_connector.callback_metrics
+        _cb_metrics: dict = self.logger_connector.callback_metrics
+        if not _cb_metrics and self.accelerator.training_type_plugin.distributed_backend == 'ddp_spawn':
```

Comment:
this will fail for single device since it doesn't have a `distributed_backend` attribute.

Maybe check the type instead?